app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -213,7 +213,15 @@ class PluginApi {
 
     if (canModify(klass, "member", resolverName, changes)) {
       delete changes.pluginId;
-      klass.class.reopen(changes);
+
+      if (klass.class.reopen) {
+        klass.class.reopen(changes);
+      } else {
+        Object.defineProperties(
+          klass.class.prototype || klass.class,
+          Object.getOwnPropertyDescriptors(changes)
+        );
```

Comment:
This is overwriting the existing properties, which means we can't use `super.blah` when modifying native classes.

IMO it would be very useful to have working inheritance here. I imagine it might require switching up the syntax... maybe something like

```javascript
api.modifyClass('model:user', class {
  someOverride(){
    return super.someOverride() + " suffix";
  }
});
```
And then we somehow insert that new class at the tip of the prototype chain.

Will definitely require some acrobatics to make it work, but I think it should be possible 🤞